File: Administrativo/spring-erp-servicio/net/royal/spring/minedu/dominio/dto/DtoContratacion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace net.royal.spring.minedu.dominio.dto
{
    public class DtoContratacion
    {
        public Int32? IdContratacion { get; set; }
        public Int32? IdPersona { get; set; }
        public string Persona { get; set; }
        public Nullable<DateTime> Fechainicio { get; set; }
        public Nullable<DateTime> Fechafin { get; set; }
        public String IdModalidad { get; set; }
        public String Modalidad { get; set; }
        public String Numeroorden { get; set; }
        public String IdCargo { get; set; }
        public String Cargo { get; set; }
        public String Conocimientos { get; set; }
        public String Estado { get; set; }
        public Int32? Secuencia { get; set; }
        public Int32? Plazo { get; set; }

        [NotMapped]
        public List<DateTime?> diasocupados;

        public DtoContratacion()
        {
            diasocupados = new List<DateTime?>();
        }
    }
}

File: Administrativo/spring-erp-web/SpringNg/src/app/erp_module/minedu/contratacion/dominio/dtoContratacion.ts
import { dtoFechasdisponibles } from "./dtoFechasdisponibles";
import { Proyectorecursoperiodo } from "../../proyecto/dominio/proyectorecursoperiodo";

export class dtoContratacion {
    idContratacion: number;
    persona: string;
    fechainicio: Date;
    fechafin: Date;
    idModalidad: string;
    modalidad: string;
    numeroorden: string;
    idCargo: string;
    cargo: string;
    estado: string;
    idPersona: number;

    creacionUsuario: string;
    creacionFecha: Date;
    creacionTerminal: string;
    modificacionUsuario: string;
    modificacionFecha: Date;
    modificacionTerminal: string;

    conocimientos: string;
    secuencia: number;

    //parametors no mapeados
    diasocupados: Date[];
    rango: Date[];
    rol: string;
    horas: number;

    listaPeriodos: Proyectorecursoperiodo[];

    constructor() {
        this.listaPeriodos = [];
    }


}